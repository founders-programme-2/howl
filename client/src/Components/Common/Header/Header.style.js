import styled from 'styled-components';
import { Link } from 'react-router-dom';
import responsiveSizes from '../../../constants/responsiveSizes';

const { SML, LRG } = responsiveSizes;

export const NavSection = styled.nav`
  background-color: black;
  display: flex;
  flex-flow: row nowrap;
  min-width: 100%;

  @media (max-width: ${SML}) {
    flex-flow: column nowrap;
  }
`;

export const NavList = styled.ul`
  list-style: none;
  display: flex;
  flex-flow: row nowrap;
  margin: auto 2% auto 2%;
  padding: 0;

  @media (max-width: ${LRG}) {
    flex-flow: row wrap;
    justify-content: space-around;
  }

  @media (max-width: ${SML}) {
    flex-flow: column nowrap;
  }
`;

export const NavLi = styled.li`
  text-align: center;
  padding: 1.5rem;
`;

export const NavLink = styled(Link)`
  font-family: 'Raleway', sans-serif;
  font-weight: 800;
  font-size: 1.3rem;
  color: ${props => props.theme.color};
  transition: all 0.2s ease-in-out;
  text-decoration: none;
  text-transform: uppercase;
  letter-spacing: 0.05em;

  &:hover {
    color: var(--main-btn-color);
    transform: scale(1.1);
  }
`;

export const Logo = styled.img`
  margin: 1rem 1rem 0 1rem;
  transition: all 0.2s ease-in-out;
  &:hover {
    transform: scale(1.1);
  }
`;

export const NavBtn = styled.button`
  text-transform: uppercase;
  transition: all 0.2s ease-in-out;
  font-size: 1.2rem;

  &:hover {
    cursor: pointer;
    background-color: var(--main-bg-color);
    transform: scale(1.1);
  }
`;

// Had to wrap button/logo in a div in order for the margin to work. Fragments did not work.
export const CntrdWrapper = styled.div`
  margin: auto 5% auto 5%;

  @media (max-width: ${LRG}) {
    margin: auto;
  }

  @media (max-width: ${SML}) {
    margin: auto;
  }
`;

export const Sa = styled.a`
  font-size: 0;
`;
