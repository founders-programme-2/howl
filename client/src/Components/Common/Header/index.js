import React from 'react';
import { Link, withRouter } from 'react-router-dom';
import navigationUrls from '../../../constants/navigationUrls';

import {
  NavLink,
  NavList,
  NavSection,
  NavLi,
  Logo,
  NavBtn,
  CntrdWrapper,
  Sa,
} from './Header.style';
import logoPath from '../../../assets/logo.svg';

const {
  ABOUT_URL,
  ADD_URL,
  ARCHIVE_URL,
  RESOURCES_URL,
  TIMELINE_URL,
  WLM_URL,
} = navigationUrls;

const Header = ({ location }) => {
  const { pathname: url } = location;
  return (
    <NavSection>
      <Sa href="#top" name="topLink" aria-label="Navigation to top of page">
        Top of page
      </Sa>
      <CntrdWrapper>
        <Link to="/">
          <Logo src={logoPath} alt="Howl logo" />
        </Link>
      </CntrdWrapper>

      <NavList>
        <NavLi>
          <NavLink
            to={ABOUT_URL}
            theme={{
              color: url === ABOUT_URL ? 'var(--main-btn-color)' : 'white',
            }}
          >
            About Howl
          </NavLink>
        </NavLi>

        <NavLi>
          <NavLink
            to={WLM_URL}
            theme={{
              color: url === WLM_URL ? 'var(--main-btn-color)' : 'white',
            }}
          >
            Women&#39;s Liberation
          </NavLink>
        </NavLi>

        <NavLi>
          <NavLink
            to={ARCHIVE_URL}
            theme={{
              color: url === ARCHIVE_URL ? 'var(--main-btn-color)' : 'white',
            }}
          >
            The Archive
          </NavLink>
        </NavLi>

        <NavLi>
          <NavLink
            to={TIMELINE_URL}
            theme={{
              color: url === TIMELINE_URL ? 'var(--main-btn-color)' : 'white',
            }}
          >
            Interactive Timeline
          </NavLink>
        </NavLi>

        <NavLi>
          <NavLink
            to={ADD_URL}
            theme={{
              color: url === RESOURCES_URL ? 'var(--main-btn-color)' : 'white',
            }}
          >
            Add Story
          </NavLink>
        </NavLi>

        <NavLi>
          <NavLink
            to={RESOURCES_URL}
            theme={{
              color: url === RESOURCES_URL ? 'var(--main-btn-color)' : 'white',
            }}
          >
            Feminist Resources
          </NavLink>
        </NavLi>
      </NavList>
    </NavSection>
  );
};

export default withRouter(Header);
