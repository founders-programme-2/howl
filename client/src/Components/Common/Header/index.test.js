import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Header from './index';

test('Natural', () => {
  const header = renderer
    .create(
      <BrowserRouter>
        {' '}
        <Header />{' '}
      </BrowserRouter>
    )
    .toJSON();
  expect(header).toMatchSnapshot();
});
