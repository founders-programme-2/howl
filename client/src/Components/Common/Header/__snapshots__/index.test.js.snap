// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Natural 1`] = `
Array [
  " ",
  <nav
    className="sc-bdVaJa bnmdUt"
  >
    <div
      className="sc-EHOje dWTUpW"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Howl logo"
          className="sc-ifAKCX iZOrPs"
          src="logo.svg"
        />
      </a>
    </div>
    <ul
      className="sc-bwzfXH hPfERK"
    >
      <li
        className="sc-htpNat kdVOZR"
      >
        <a
          className="sc-bxivhb bRkUzh"
          href="/about"
          onClick={[Function]}
          theme={
            Object {
              "color": "white",
            }
          }
        >
          About Howl
        </a>
      </li>
      <li
        className="sc-htpNat kdVOZR"
      >
        <a
          className="sc-bxivhb bRkUzh"
          href="/wlm"
          onClick={[Function]}
          theme={
            Object {
              "color": "white",
            }
          }
        >
          Women's Liberation
        </a>
      </li>
      <li
        className="sc-htpNat kdVOZR"
      >
        <a
          className="sc-bxivhb bRkUzh"
          href="/timeline"
          onClick={[Function]}
          theme={
            Object {
              "color": "white",
            }
          }
        >
          Interactive Timeline
        </a>
      </li>
      <li
        className="sc-htpNat kdVOZR"
      >
        <a
          className="sc-bxivhb bRkUzh"
          href="/archive"
          onClick={[Function]}
          theme={
            Object {
              "color": "white",
            }
          }
        >
          Our Stories
        </a>
      </li>
      <li
        className="sc-htpNat kdVOZR"
      >
        <a
          className="sc-bxivhb bRkUzh"
          href="/add"
          onClick={[Function]}
          theme={
            Object {
              "color": "white",
            }
          }
        >
          Add Story
        </a>
      </li>
      <li
        className="sc-htpNat kdVOZR"
      >
        <a
          className="sc-bxivhb bRkUzh"
          href="/resources"
          onClick={[Function]}
          theme={
            Object {
              "color": "white",
            }
          }
        >
          Feminist Resources
        </a>
      </li>
    </ul>
  </nav>,
  " ",
]
`;
